src/libsyntax/tokenstream.rs

```diff
@@ -241,6 +258,21 @@ impl TokenStream {
         }
     }
 
+    pub fn map_pos<F: FnMut(usize, TokenTree) -> TokenTree>(self, mut f: F) -> TokenStream {
```

Comment:
I'd rename this into `enumerate_map` (or `map_enumerated`), to use the same naming conventions as `Iterator`.